internal/pkg/agent/application/upgrade/rollback.go

```diff
@@ -46,7 +46,7 @@ func Rollback(ctx context.Context, log *logger.Logger, c client.Client, topDirPa
 		symlinkTarget = paths.BinaryPath(filepath.Join(paths.DataFrom(topDirPath), hashedDir), agentName)
 	}
 	// change symlink
-	if err := changeSymlinkInternal(log, topDirPath, symlinkPath, symlinkTarget); err != nil {
+	if err := ChangeSymlink(log, topDirPath, symlinkPath, symlinkTarget); err != nil {
```

Comment:
I think this could be left as-is or even just renamed to `changeSymlink`.  IOW, it doesn't look like this function is called from outside the `upgrade` package so no need to export it.

Reply:
280202026ca9bd5b2c78412f70ba5e77cd27f474